tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp

```diff
@@ -688,7 +697,7 @@ bool ASTProducer::shouldRebuild(SwiftASTManager::Implementation &MgrImpl,
       InputStamps.push_back(MgrImpl.getBufferStamp(File));
   }
   assert(InputStamps.size() ==
-         Invok.Opts.Invok.getFrontendOptions().Inputs.inputFilenameCount());
+         Invok.Opts.Invok.getFrontendOptions().Inputs.inputCount());
```

Comment:
If we have non-file inputs, this assertion would fail, so either we should assert above that all our inputs are files or here we need to get the right count.

Reply:
With refactoring every input is a "file", so the assertion cannot fail. But the test for an empty file name can never fail, so I'll remove it.

Reply:
Should it be an assertion after getFile() then?

Reply:
Good idea--better yet, I added assertions to the constructor and to the accessor.